deploy/lambda/auto-deployment.js

```diff
@@ -33,33 +33,50 @@ exports.handler = async function (event, context) {
                     "type": "pipeline_ref_target",
                     "ref_type": "branch",
                     "ref_name": "master",
-                    "selector": {"type": "custom", "pattern": "deploy-to-prod"}
+                    "selector": { "type": "custom", "pattern": "deploy-to-prod" }
                 }
             };
 
             const pipelineResponse = await postRequest(
-              pipelinesURL,
-              body,
-              {'Authorization': 'Basic ' + new Buffer(autoDeployUsername + ':' + autoDeployToken).toString('base64')}
+                pipelinesURL,
+                body,
+                { 'Authorization': 'Basic ' + new Buffer(autoDeployUsername + ':' + autoDeployToken).toString('base64') }
             );
-            console.log("Trigerring Pipeline API Response ", pipelineResponse);
             if (pipelineResponse.type === 'pipeline') {
-                console.log('Pipeline triggered successfully!');
+                console.log(`Build# ${pipelineResponse?.build_number} triggered successfully!`);
             } else {
                 throw new Error(JSON.stringify(pipelineResponse));
             }
         } else {
-            console.log('No changes found, skipping deployment!');
+            console.log('No changes found or pipeline already running, skipping deployment!');
         }
     } catch (e) {
         console.error('Error: ', e);
     }
 };
 
-function getRequest(url) {
+// Returns true if any pipline is running for deployment to stage/prod
+async function isPipelineRunning() {
+    const url = "https://api.bitbucket.org/2.0/repositories/atlassian/github-for-jira-deployment/pipelines/?page=1&pagelen=20&sort=-created_on"
+    const pipelines = await getRequest(url, {
+        'Authorization': 'Basic ' + new Buffer('fusion-release-bot:2nXHekMCpxc3T8rGPKwP').toString('base64')
```

Comment:
The `username` and `password` here needs to be saved in Microscope. This will trigger a vulnerability issue.

Reply:
> The `username` and `password` here needs to be saved in Microscope. This will trigger a vulnerability issue.

Thanks kay, update it. I put it for local testing.